code/controllers/subsystem/fluids.dm

```diff
@@ -0,0 +1,266 @@
+// Flags indicating what parts of the fluid the subsystem processes.
+/// Indicates that a fluid subsystem processes fluid spreading.
+#define SS_PROCESSES_SPREADING (1<<0)
+/// Indicates that a fluid subsystem processes fluid effects.
+#define SS_PROCESSES_EFFECTS (1<<1)
+
+/**
+ * # Fluid Subsystem
+ *
+ * A subsystem that processes the propagation and effects of a particular fluid.
+ *
+ * Both fluid spread and effect processing are handled through a carousel system.
+ * Fluids being spread and fluids being processed are organized into buckets.
+ * Each fresh (non-resumed) fire one bucket of each is selected to be processed.
+ * These selected buckets are then fully processed.
+ * The next fresh fire selects the next bucket in each set for processing.
+ * If this would walk off the end of a carousel list we wrap back to the first element.
+ * This effectively makes each set a circular list, hence a carousel.
+ */
+SUBSYSTEM_DEF(fluids)
+	name = "Fluid"
+	wait = 0.1 SECONDS
+	flags = SS_BACKGROUND|SS_KEEP_TIMING
+	runlevels = RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+
+	/// What this subsystem processes. Yes, it's called 'case' entirely because of 'switch() case: '.
+	var/case = NONE
+
+	// Fluid spread processing:
+	/// The amount of time before a fluid node is created and when it spreads.
+	var/spread_wait = 1 SECONDS
+	/// The number of buckets in the spread carousel.
+	var/tmp/num_spread_buckets
```

Comment:
`/tmp` exists for save file reasons, and isn't something we use as a syntax hint. shouldn't be used here

Reply:
Alright.

Reply:
Done.